R/Ecdc.R

```diff
@@ -0,0 +1,122 @@
+#' R6 Class containing specific attributes and methods for ECDC dataset
+#'
+#' @description Country specific information for downloading, cleaning
+#'  and processing covid-19 region data for ECDC
+#'
+#' @details Inherits from `dataClass`
+#'
+Ecdc <- R6::R6Class("ecdc",
+  inherit = dataClass,
+  public = list(
+
+    # Core Attributes
+    #' @field level_1_region the level 1 region name.
+    level_1_region = "country",
+    #' @field level_2_region the level 2 region name.
+    level_2_region = NA, # If no level 2 regions, set as NA
+    #' @field data_url link to raw data
+    data_url = "https://opendata.ecdc.europa.eu/covid19/casedistribution/csv",
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "deaths_new"),
+
+    #' @description ECDC specific state level data cleaning
+    #' @param ... pass additional arguments
+    #' @importFrom dplyr mutate rename select arrange filter
+    #' @importFrom stringr str_replace_all
+    #' @importFrom countrycode countryname countrycode
+    #'
+    clean_regional = function(...) {
```

Comment:
can still call them clean, the data rets stored as a variable, of which clean is an attribute. e.g. Italy has a method clean (Italy$clean) and has data called region of which clean is an attribute (region$clean). 

It could get confusing but the names should still work ok.

Reply:
ah sweet that sounds good.